sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollResponse.java

```diff
@@ -37,35 +37,45 @@
      * {@code USER_CANCELLED} or {@code FAILED}.
      */
     public static final class OperationStatus extends ExpandableStringEnum<OperationStatus> {
+
+        private boolean completed;
```

Comment:
Is this necessary? can it be inferred from OperationStatus?

Reply:
In Processor not? it seems inferred from your usages.

Reply:
I'm curious why not? it seems like you're inferring it from those status' below.

Reply:
if some user supplies 'TOTALLY_COMPLETED' as final status, then we do not know that is the final status.
Poller.isComplete relies on this boolean too now to signal is Poller completed or not.

Reply:
Thanks for clarifying!